### PR TITLE
Document VSTest Bridge extension

### PR DESCRIPTION
**Pull request by Evangelink**
## Summary

Split Microsoft Testing Platform extensions per category pages and add the VSTest Bridge documentation

cc @MarcoRossignoli @nohwnd 

---

#### Internal previews

<details><summary><strong>Toggle expand/collapse</strong></summary><br/>

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/testing/unit-testing-platform-exit-codes.md](https://github.com/dotnet/docs/blob/b19cdf5d0c6f15dce9d4bebb58caf19405232638/docs/core/testing/unit-testing-platform-exit-codes.md) | [Microsoft.Testing.Platform exit codes](https://review.learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-exit-codes?branch=pr-en-us-40380) |
| [docs/core/testing/unit-testing-platform-extensions-code-coverage.md](https://github.com/dotnet/docs/blob/b19cdf5d0c6f15dce9d4bebb58caf19405232638/docs/core/testing/unit-testing-platform-extensions-code-coverage.md) | [Microsoft.Testing.Platform Code Coverage extensions](https://review.learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-extensions-code-coverage?branch=pr-en-us-40380) |
| [docs/core/testing/unit-testing-platform-extensions-diagnostics.md](https://github.com/dotnet/docs/blob/b19cdf5d0c6f15dce9d4bebb58caf19405232638/docs/core/testing/unit-testing-platform-extensions-diagnostics.md) | [Diagnostics extensions](https://review.learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-extensions-diagnostics?branch=pr-en-us-40380) |
| [docs/core/testing/unit-testing-platform-extensions-hosting.md](https://github.com/dotnet/docs/blob/b19cdf5d0c6f15dce9d4bebb58caf19405232638/docs/core/testing/unit-testing-platform-extensions-hosting.md) | [docs/core/testing/unit-testing-platform-extensions-hosting](https://review.learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-extensions-hosting?branch=pr-en-us-40380) |
| [docs/core/testing/unit-testing-platform-extensions-policy.md](https://github.com/dotnet/docs/blob/b19cdf5d0c6f15dce9d4bebb58caf19405232638/docs/core/testing/unit-testing-platform-extensions-policy.md) | [Policy extensions](https://review.learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-extensions-policy?branch=pr-en-us-40380) |
| [docs/core/testing/unit-testing-platform-extensions-test-reports.md](https://github.com/dotnet/docs/blob/b19cdf5d0c6f15dce9d4bebb58caf19405232638/docs/core/testing/unit-testing-platform-extensions-test-reports.md) | [Microsoft.Testing.Platform test reports extensions](https://review.learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-extensions-test-reports?branch=pr-en-us-40380) |
| [docs/core/testing/unit-testing-platform-extensions-vstest-bridge.md](https://github.com/dotnet/docs/blob/b19cdf5d0c6f15dce9d4bebb58caf19405232638/docs/core/testing/unit-testing-platform-extensions-vstest-bridge.md) | [VSTest Bridge extension](https://review.learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-extensions-vstest-bridge?branch=pr-en-us-40380) |
| [docs/core/testing/unit-testing-platform-extensions.md](https://github.com/dotnet/docs/blob/b19cdf5d0c6f15dce9d4bebb58caf19405232638/docs/core/testing/unit-testing-platform-extensions.md) | [docs/core/testing/unit-testing-platform-extensions](https://review.learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-extensions?branch=pr-en-us-40380) |
| [docs/core/testing/unit-testing-platform-integration-dotnet-test.md](https://github.com/dotnet/docs/blob/b19cdf5d0c6f15dce9d4bebb58caf19405232638/docs/core/testing/unit-testing-platform-integration-dotnet-test.md) | [Use Microsoft.Testing.Platform with `dotnet test`](https://review.learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-platform-integration-dotnet-test?branch=pr-en-us-40380) |
| [docs/whats-new/dotnet-docs-mod0.md](https://github.com/dotnet/docs/blob/b19cdf5d0c6f15dce9d4bebb58caf19405232638/docs/whats-new/dotnet-docs-mod0.md) | [".NET docs: What's new for March 2024"](https://review.learn.microsoft.com/en-us/dotnet/whats-new/dotnet-docs-mod0?branch=pr-en-us-40380) |
| [docs/whats-new/dotnet-docs-mod1.md](https://github.com/dotnet/docs/blob/b19cdf5d0c6f15dce9d4bebb58caf19405232638/docs/whats-new/dotnet-docs-mod1.md) | [.NET docs: What's new for January 2024](https://review.learn.microsoft.com/en-us/dotnet/whats-new/dotnet-docs-mod1?branch=pr-en-us-40380) |

</details>

### CODE REVIEW

**Review by IEvangelist — APPROVED** [review]

File: docs/core/testing/unit-testing-platform-extensions-code-coverage.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+## Microsoft code coverage
+
+Microsoft Code Coverage analysis is possible for both managed (CLR) and unmanaged (native) code. Both static and dynamic instrumentation are supported. This extension is shipped as part of [Microsoft.Testing.Extensions.CodeCoverage](https://nuget.org/packages/Microsoft.Testing.Extensions.CodeCoverage) NuGet package.
+
+> [!NOTE]
+> Unmanaged (native) code coverage is disabled in the extension by default. Use flags `EnableStaticNativeInstrumentation` and `EnableDynamicNativeInstrumentation` to enable it if needed.
+> For more information about unmanaged code coverage, see [Static and dynamic native instrumentation](/visualstudio/test/customizing-code-coverage-analysis?#static-and-dynamic-native-instrumentation).
+
+> [!NOTE]
+> The package is shipped with Microsoft .NET LIBRARY closed-source free to use licensing model.
```
Comment (IEvangelist) [inline docs/core/testing/unit-testing-platform-extensions-code-coverage.md:28]:
```suggestion
> [!IMPORTANT]
> The package is shipped with Microsoft .NET library closed-source free to use licensing model.
```

File: docs/core/testing/unit-testing-platform-extensions-diagnostics.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+> [!NOTE]
+> Environment variables take precedence over the command line arguments.
+
+## Crash dump
+
+This extension allows you to create a crash dump file if the process crashes. This extension is shipped as part of [Microsoft.Testing.Extensions.CrashDump](https://nuget.org/packages/Microsoft.Testing.Extensions.CrashDump) NuGet package.
+
+> [!NOTE]
+> The package is shipped with Microsoft .NET LIBRARY closed-source free to use licensing model.
```
Comment (IEvangelist) [inline docs/core/testing/unit-testing-platform-extensions-diagnostics.md:42]:
```suggestion
> [!IMPORTANT]
> The package is shipped with Microsoft .NET library closed-source free to use licensing model.
```

File: docs/core/testing/unit-testing-platform-extensions-diagnostics.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+> [!CAUTION]
+> The extension isn't compatible with .NET Framework and will be silently ignored. For .NET Framework support, you enable the postmortem debugging with Sysinternals ProcDump. For more information, see [Enabling Postmortem Debugging: Window Sysinternals ProcDump](/windows-hardware/drivers/debugger/enabling-postmortem-debugging#window-sysinternals-procdump). The postmortem debugging solution will also collect process crash information for .NET so you can avoid the use of the extension if you're targeting both .NET and .NET Framework test applications.
+
+## Hang dump
+
+This extension allows you to create a dump file after a given timeout. This extension is shipped as part of [Microsoft.Testing.Extensions.HangDump](https://nuget.org/packages/Microsoft.Testing.Extensions.HangDump) package.
+
+> [!NOTE]
+> The package is shipped with Microsoft .NET LIBRARY closed-source free to use licensing model.
```
Comment (IEvangelist) [inline docs/core/testing/unit-testing-platform-extensions-diagnostics.md:60]:
```suggestion
> [!IMPORTANT]
> The package is shipped with Microsoft .NET library closed-source free to use licensing model.
```

File: docs/core/testing/unit-testing-platform-extensions-test-reports.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+This article list and explains all `Microsoft Testing Platform` extensions related to the test report capability.
+
+A test report is a file that contains information about the execution and outcome of the tests.
+
+## Visual Studio test reports
+
+The Visual Studio test result file (or TRX) is the default format for publishing test results. This extension is shipped as part of [Microsoft.Testing.Extensions.TrxReport](https://nuget.org/packages/Microsoft.Testing.Extensions.TrxReport) package.
+
+> [!NOTE]
+> The package is shipped with Microsoft .NET LIBRARY closed-source free to use licensing model.
```
Comment (IEvangelist) [inline docs/core/testing/unit-testing-platform-extensions-test-reports.md:20]:
```suggestion
> [!IMPORTANT]
> The package is shipped with Microsoft .NET library closed-source free to use licensing model.
```

File: docs/core/testing/unit-testing-platform-extensions-vstest-bridge.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+ms.author: amauryleve
+ms.date: 04/10/2024
+---
+
+# VSTest Bridge extension
+
+This extension provides a compatibility layer with VSTest allowing the test frameworks depending on it to continue supporting running in VSTest mode (`vstest.console.exe`, usual `dotnet test`, `VSTest task` on AzDo, Test Explorers of Visual Studio and Visual Studio Code...). This extension is shipped as part of [Microsoft.Testing.Extensions.VSTestBridge](https://www.nuget.org/packages/Microsoft.Testing.Extensions.VSTestBridge) package.
+
+> [!NOTE]
+> The package is shipped with Microsoft .NET LIBRARY closed-source free to use licensing model.
```
Comment (IEvangelist) [inline docs/core/testing/unit-testing-platform-extensions-vstest-bridge.md:14]:
```suggestion
> [!IMPORTANT]
> The package is shipped with Microsoft .NET library closed-source free to use licensing model.
```

**Comment by IEvangelist** [comment]
LGTM, but there are a few build errors: 
image

**Comment by Evangelink** [comment]
Oh right sorry, I will fix them tomorrow.

**Comment by Evangelink** [comment]
@IEvangelist I don't know what's the recommended action for the "what's new" pages when the article has been moved. Shall I change the path?

**Comment by Evangelink** [comment]
Thanks! I wasn't sure it was ok to edit past "what's new" articles.

**Pull request merged**